jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/FormFields.java

```diff
@@ -208,12 +206,16 @@ public static void onFields(Request request, Charset charset, Promise.Invocable<
      *
      * @param request The request to get or read the Fields from
      * @param charset The {@link Charset} of the request content, if previously extracted.
-     * @param maxFields The maximum number of fields to accept
-     * @param maxLength The maximum length of fields
+     * @param maxFields The maximum number of fields to accept; or -1 for a default
+     * @param maxLength The maximum length of fields; or -1 for a default
```

Comment:
This is a bit misleading in a way, I think it was clearer before.

Because you can configure these to be -1 for no limit on the length.

So now if the context attribute of `MAX_LENGTH_ATTRIBUTE` was set to -1, then it is unlimited, but if you call this method with -1 it is limited to `MAX_LENGTH_DEFAULT`.

Reply:
I agree it is a bit confusing, but I need a way to pass in "use default value", otherwise we end up encoding default values in XML (see jetty-eager-content.xml) and in java.  Do you have an alternative suggestion?

Reply:
@lachlan-roberts and finally... the other API is deprecated, so it will soon go away.   You can pass in MAX_VALUE for unlimited.